src/lib.rs

```diff
@@ -62,30 +63,46 @@ impl Decoder {
             }
 
             Decoder {
-                dec: dec.assume_init(),
+                dec: ptr::NonNull::new(dec.assume_init()).unwrap(),
+                pending_data: None,
             }
         }
     }
 
-    pub fn flush(&self) {
+    pub fn flush(&mut self) {
         unsafe {
-            dav1d_flush(self.dec);
+            dav1d_flush(self.dec.as_ptr());
+            if let Some(mut pending_data) = self.pending_data.take() {
+                dav1d_data_unref(&mut pending_data);
+            }
         }
     }
 
-    pub fn send_data<T: AsRef<[u8]>>(
+    pub fn send_data<T: AsRef<[u8]> + Send + 'static>(
```

Comment:
`Send` is also needed because the input data might be passed to other threads by the decoder when it does multi-threaded decoding.

Reply:
Agreed